<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Show method for phyloCompData object — show,phyloCompData-method • phylocompcodeR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Show method for phyloCompData object — show,phyloCompData-method" />
<meta property="og:description" content="Show method for phyloCompData object." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



<!-- dependencies from examples -->


  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phylocompcodeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.27.3.9999</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/compcodeR.html">Comparing methods for differential expression analysis of RNAseq data with the compcodeR package</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/csoneson/compcodeR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Show method for <code>phyloCompData</code> object</h1>
    <small class="dont-index">Source: <a href='https://github.com/csoneson/compcodeR/blob/master/R/show-methods.R'><code>R/show-methods.R</code></a></small>
    <div class="hidden name"><code>show-phyloCompData-method.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Show method for <code>phyloCompData</code> object.</p>
    </div>

    <pre class="usage"><span class='co'># S4 method for phyloCompData</span>
<span class='fu'><a href='https://rdrr.io/pkg/TCC/man/TCC-class.html'>show</a></span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>A <code>phyloCompData</code> object</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Charlotte Soneson, Paul Bastide</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>mydata</span> <span class='op'>&lt;-</span> <span class='fu'><a href='generateSyntheticData.html'>generateSyntheticData</a></span><span class='op'>(</span>dataset <span class='op'>=</span> <span class='st'>"mydata"</span>, n.vars <span class='op'>=</span> <span class='fl'>1000</span>, 
                                samples.per.cond <span class='op'>=</span> <span class='fl'>5</span>, n.diffexp <span class='op'>=</span> <span class='fl'>100</span>,
                                id.species <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>10</span><span class='op'>)</span>,
                                tree <span class='op'>=</span> <span class='fu'>ape</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ape/man/rlineage.html'>rphylo</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span>,
                                lengths.relmeans <span class='op'>=</span> <span class='st'>"auto"</span>, lengths.dispersions <span class='op'>=</span> <span class='st'>"auto"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: `id.species` is not named. I'm naming them, assuming they are in the same order as the tree.</span></div><div class='input'><span class='va'>mydata</span>
</div><div class='output co'>#&gt; An object of class phyloCompData 
#&gt; Dataset name: mydata 
#&gt; Number of samples: 10 
#&gt; Number of variables: 1000 
#&gt; Number of random outliers: 0 
#&gt; Number of single outliers: 0 
#&gt; Number of known truly differentially expressed genes: 100 
#&gt; No differential expression analysis has been performed. 
#&gt; 
#&gt; count.matrix:
#&gt;       t1    t2    t3    t4    t5    t6
#&gt; g1   677   464   623   540   994  1244
#&gt; g2   212    91   136   469    38   320
#&gt; g3  8650  3501  5651 10353  5132 20182
#&gt; g4  2244   918  1921   786  3791 12941
#&gt; g5  1408  2153  3537  2927  1555  4067
#&gt; g6 19796 23016 32019 12395 49297 34190
#&gt; + 994 rows and 4 cols...
#&gt; 
#&gt; sample.annotations:
#&gt;    condition depth.factor id.species id.condition
#&gt; t1         1    1.1059100          1            1
#&gt; t2         1    0.8032296          2            1
#&gt; t3         1    1.1182325          3            1
#&gt; t4         1    1.1147645          4            1
#&gt; t5         1    1.3286844          5            1
#&gt; t6         2    0.8657358          6            2
#&gt; + 4 rows...
#&gt; 
#&gt; variable.annotations:
#&gt;    truedispersions.S1 truedispersions.S2 truemeans.S1 truemeans.S2
#&gt; g1         0.06746282         0.06746282     91.07018    186.80398
#&gt; g2         0.35363024         0.35363024     20.24390     41.06688
#&gt; g3         0.10650664         0.10650664    249.17544    793.82092
#&gt; g4         0.22208438         0.22208438     39.14634    206.19044
#&gt; g5         0.11717587         0.11717587    229.14634    504.91998
#&gt; g6         0.08663472         0.08663472    464.22807    933.00555
#&gt;    n.random.outliers.up.S1 n.random.outliers.up.S2 n.random.outliers.down.S1
#&gt; g1                       0                       0                         0
#&gt; g2                       0                       0                         0
#&gt; g3                       0                       0                         0
#&gt; g4                       0                       0                         0
#&gt; g5                       0                       0                         0
#&gt; g6                       0                       0                         0
#&gt;    n.random.outliers.down.S2 n.single.outliers.up.S1 n.single.outliers.up.S2
#&gt; g1                         0                       0                       0
#&gt; g2                         0                       0                       0
#&gt; g3                         0                       0                       0
#&gt; g4                         0                       0                       0
#&gt; g5                         0                       0                       0
#&gt; g6                         0                       0                       0
#&gt;    n.single.outliers.down.S1 n.single.outliers.down.S2   M.value   A.value
#&gt; g1                         0                         0 0.9084921  9.659980
#&gt; g2                         0                         0 1.5140718  7.742311
#&gt; g3                         0                         0 1.7738109 13.371876
#&gt; g4                         0                         0 2.1714704 11.652939
#&gt; g5                         0                         0 1.1464079 11.550079
#&gt; g6                         0                         0 0.6879611 14.816299
#&gt;    truelog2foldchanges upregulation downregulation differential.expression
#&gt; g1            1.036475            1              0                       1
#&gt; g2            1.020488            1              0                       1
#&gt; g3            1.671652            1              0                       1
#&gt; g4            2.397028            1              0                       1
#&gt; g5            1.139786            1              0                       1
#&gt; g6            1.007052            1              0                       1
#&gt;    lengths.relmeans lengths.dispersions M.value.TPM A.value.TPM
#&gt; g1         1048.357         0.002140625   0.9137434  -0.3109954
#&gt; g2         3344.571         1.012137002   0.8081918  -2.6379876
#&gt; g3         5960.214         0.295973014   1.4061421   1.3670502
#&gt; g4         5180.857         0.239087670   2.4666154  -0.6862882
#&gt; g5         3419.000         0.468503762   1.0678868   0.7691664
#&gt; g6         6651.714         0.257245838   1.1252715   1.7610089
#&gt; + 994 rows...
#&gt; 
#&gt; Phylogenetic tree:
#&gt; 
#&gt; Phylogenetic tree with 10 tips and 9 internal nodes.
#&gt; 
#&gt; Tip labels:
#&gt;   t1, t2, t3, t4, t5, t6, ...
#&gt; 
#&gt; Rooted; includes branch lengths.
#&gt; 
#&gt; length.matrix:
#&gt;      t1    t2    t3   t4    t5   t6
#&gt; g1 1017   966   969 1004  1033  948
#&gt; g2 1352   797   675 3133   522 1478
#&gt; g3 4357  2609  3214 5569  2928 5110
#&gt; g4 5965  5802  6721 3340  7706 8664
#&gt; g5 1001  2301  2408 2868   889 1723
#&gt; g6 8443 10457 10574 5099 19246 7769
#&gt; </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Charlotte Soneson.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


